src/helics/shared_api_library/helics.h

```diff
@@ -764,7 +764,7 @@ HELICS_EXPORT void
  * @param[in,out] err An error object that will contain an error code and string if any error occurred during the execution of the function.
  * @endforcpponly
  */
-HELICS_EXPORT void helicsFederateInfoSetIntegerProperty(helics_federate_info fi, int intProperty, int propertyValue, helics_error* err);
+HELICS_EXPORT void helicsFederateInfoSetIntegerProperty(helics_federate_info fi, helics_properties intProperty, int propertyValue, helics_error* err);
```

Comment:
Is `helics_properties` a subset of what can be set over here? This should be reviewed.

Reply:
The biggest issue is that if we are (eventually) to allow user defined core-types, then we need to allow user defined properties on those core types.